packages/shared/src/styles/components/buttons.css

```
@@ -135,6 +134,7 @@
       --button-border-color: var(--button-hover-border-color);
       --button-color: var(--button-hover-color);
       --button-box-shadow: var(--button-hover-box-shadow);
+      transition: all 0.2s ease-in-out;
```

Just to any unwanted behaviors, I think it would be best if we specify the only part where we are trying to make changes, which is the `background-color`.
```suggestion
      transition: background-color 0.2s ease-in-out;
```